manifests/kustomize/base/installs/generic/pipeline-install-config.yaml

```diff
@@ -68,6 +68,19 @@ data:
   ## any node and avoid defaulting to specific nodes. Allowed values are:
   ## 'false' and 'true'.
   cacheNodeRestrictions: "false"
+  ## MAXIMUM_CACHE_STALENESS configures caching according to
+  ## https://www.kubeflow.org/docs/components/pipelines/overview/caching/ and
+  ## https://www.kubeflow.org/docs/components/pipelines/overview/caching-v2/.
+  ## Larger than MAXIMUM_CACHE_STALENESS per pipeline user set values are 
+  ## reduced to MAXIMUM_CACHE_STALENESS.
+  ## The administrator of the storage backend can rely on it to delete old cache
+  ## artifacts.
+  MAXIMUM_CACHE_STALENESS: "P30D"
+  ## DEFAULT_CACHE_STALENESS configures caching according to
+  ## https://www.kubeflow.org/docs/components/pipelines/overview/caching/ and
```

Comment:
According to this doc, default is infinity, so this is a change of behavior, right?

> By default the `max_cache_staleness` is set to infinity so any old cached data will be reused.
https://www.kubeflow.org/docs/components/pipelines/overview/caching/

Reply:
There are two options:

1. workaround to have -1 by default. for both default and maximum cache staleness.
2.  I update the documentation. I think this is an overdue change, because infinity was a bad value to start with. This is exactly why so many companies have trouble with caching and storage limits. 

My stuff is at the end of the design document https://docs.google.com/document/d/1WdPjTgoDj5AdPNYVzqYwVIWKsWjPPphYD3Ck4DFxSLA/edit#

Reply:
@chensun @Linchin i am looking forward to implement whatever option you choose.

Reply:
I think option 2 makes a lot of sense, but @chensun may have more context on this.

Reply:
As discussed in the nightly KFP meeting I am supposed to not change the documentation and have infinity cache staleness (-1) as before.
I tried just commenting out the keys in the pipeline-install configmap. This errors on the kubernetes level with Error: couldn't find key MAXIMUM_CACHE_STALENESS in ConfigMap kubeflow/pipeline-install-config. So i will comment out MAXIMUM_CACHE_STALENESS: "P30D" and DEFAULT_CACHE_STALENESS: "P7D" in the pipeline install configmap and add MAXIMUM_CACHE_STALENESS: "" and DEFAULT_CACHE_STALENESS: "" above those comments. So the administrator has a commented out example. StalenessToSeconds should already convert the empty string to -1.

```
stalenessToSeconds(staleness string) int64 {
	var seconds int64 = -1
	if d, err := duration.Parse(staleness); err == nil {
		seconds = int64(d / time.Second)
	}
	return seconds
```

I will try this after a few hours of sleep in the morning ;-)

Reply:
Infinity is working. @chensun @linchin please approve and merge.